corehq/apps/hqadmin/reports.py

```diff
@@ -170,7 +176,19 @@ def _is_limit_exceeded(self):
             start_date=self.datespan.startdate,
             end_date=self.datespan.enddate
         )
-        return NavigationEventAudit.objects.filter(**where)[:self.MAX_RECORDS + 1].count() > self.MAX_RECORDS
+        if self.selected_action in ACCESS_CHOICES:
+            where['access_type'] = self.selected_action
+            return AccessAudit.objects.filter(**where)[:self.MAX_RECORDS + 1].count() > self.MAX_RECORDS
+
+        # if action is not selected from AccessAudit, we use number of records in NavigationEventAudit as baseline
+        query = NavigationEventAudit.objects.filter(**where)
+
+        if self.selected_action:
+            query = query.extra(
+                where=["headers::jsonb->>'REQUEST_METHOD' = %s"],
+                params=[self.selected_action]
+            )
+        return query[:self.MAX_RECORDS + 1].count() > self.MAX_RECORDS
```

Comment:
This function (`_is_limit_exceeded`) now queries a different model based on the value of `self.selected_action`. Is it always used in a context where that makes sense?

For example, it is possible to get a result of `False` from querying `AccessAudit` in a context where we are actually (or also) querying `NavigationEventAudit` and therefore the number of results could be quite different?

Maybe it would make sense return a sum like `AccessAudit.objects...count() + NavigationEventAudit.objects...count()`?

Reply:
> it is possible to get a result of False from querying AccessAudit in a context where we are actually (or also) querying NavigationEventAudit

It's not. When the action is from `ACCESS_CHOICES`, we only query `AccessAudit` doc.